src/wp-includes/blocks.php

```diff
@@ -127,6 +127,68 @@ function get_block_asset_url( $path ) {
 	return plugins_url( basename( $path ), $path );
 }
 
+/**
+ * Generates a string of attributes by applying to the current block being
+ * rendered all of the features that the block supports.
+ *
+ * @since 5.6.0
+ *
+ * @param string[] $extra_attributes Optional. Array of extra attributes to render on the block wrapper.
+ * @return string String of HTML attributes.
+ */
+function get_block_wrapper_attributes( $extra_attributes = array() ) {
+	$new_attributes = WP_Block_Supports::get_instance()->apply_block_supports();
+
+	if ( empty( $new_attributes ) && empty( $extra_attributes ) ) {
+		return '';
+	}
+
+	// This is hardcoded on purpose.
+	// We only support a fixed list of attributes.
+	$attributes_to_merge = array( 'style', 'class', 'id', 'aria-label' );
+	$attributes          = array();
+	foreach ( $attributes_to_merge as $attribute_name ) {
+		$new_value   = array_key_exists( $attribute_name, $new_attributes ) ? (string) $new_attributes[ $attribute_name ] : '';
+		$extra_value = array_key_exists( $attribute_name, $extra_attributes ) ? (string) $extra_attributes[ $attribute_name ] : '';
+
+		if ( '' === $new_value && '' === $extra_value ) {
+			continue;
+		}
+
+		if ( '' === $new_value ) {
+			$attributes[ $attribute_name ] = $extra_value;
+			continue;
+		}
+
+		if ( '' === $extra_value ) {
+			$attributes[ $attribute_name ] = $new_value;
+			continue;
+		}
+
+		$attributes[ $attribute_name ] = $extra_value . ' ' . $new_value;
+	}
+
+	foreach ( $extra_attributes as $attribute_name => $value ) {
+		if ( ! in_array( $attribute_name, $attributes_to_merge, true ) ) {
+			$string_value = (string) $value;
```

Comment:
This line is also a key change introduced here, right? Shouldn't it also make sure that the value can be cast to a string? What if it is an array?